### PR TITLE
Ydoc library

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

`ydoc-server` compilation requires generation of `ydoc.cjs` resource that can take time and slow down the libraries development (building the enso distribution). This PR splits Ydoc into a library and the server part to avoid JS resources generation during the compilation of the language server.

Changelog:
- refactor: Ydoc into ~~`ydoc`~~ `ydoc-polyglot` library and  `ydoc-server` server parts
- update: language server to depend on the ~~`ydoc`~~ `ydoc-polyglot` library

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1885,13 +1886,49 @@ lazy val searcher = project
   .dependsOn(testkit % Test)
   .dependsOn(`polyglot-api`)
 
+lazy val ydoc = project
+  .in(file("lib/java/ydoc"))
```
Comment (JaroslavTulach) [inline build.sbt]: Why it is called `ydoc`? The library itself has nothing to do with `ydoc`, it is _websocket and other node.js APIs polyfill necessary to execute ydoc on JVM_.

Maybe `ydoc-polyfill`? Or `polyfills4ydoc`? Or `graal-js-polyfills`?

File: lib/java/ydoc-server/src/test/java/org/enso/ydoc/server/YdocTest.java [hunk 2]
```diff
@@ -18,15 +18,15 @@
...
-import org.enso.ydoc.jsonrpc.JsonRpcRequest;
-import org.enso.ydoc.jsonrpc.JsonRpcResponse;
-import org.enso.ydoc.jsonrpc.model.ContentRoot;
-import org.enso.ydoc.jsonrpc.model.FilePath;
-import org.enso.ydoc.jsonrpc.model.FileSystemObject;
-import org.enso.ydoc.jsonrpc.model.WriteCapability;
-import org.enso.ydoc.jsonrpc.model.result.FileListResult;
-import org.enso.ydoc.jsonrpc.model.result.InitProtocolConnectionResult;
-import org.enso.ydoc.jsonrpc.model.result.TextOpenFileResult;
+import org.enso.ydoc.server.jsonrpc.JsonRpcRequest;
```
Comment (JaroslavTulach) [inline lib/java/ydoc-server/src/test/java/org/enso/ydoc/server/YdocTest.java:21]: Wow: So many `jsonrpc` classes just for testing purposes!

I wasn't aware we have so many classes related to _mock communication with language server_. Do they _mock the communication_? What do they actually test?

**Review by JaroslavTulach — COMMENTED** [review]
I would really like to see #11477 addressed in this PR. There is one essential and one verification step:
1. modify [ContextInsightSetup](https://github.com/enso-org/enso/blob/2bbd90970547eed11b4c52f5b73c0bac785379f3/engine/common/src/main/java/org/enso/common/ContextInsightSetup.java#L170) to **enable ydoc polyfills** that are provided by the newly defined `ydoc` (as named currently) library
2. **verify** (_maybe just manually_) that the generated `ydoc.cjs` can be specified by `-Denso.dev.insight` property, loads correctly and runs the way `Ydoc` is supposed to run

Can we move forward with #11477, please?

File: build.sbt (hunk 1 above)
Comment (4e6) [reply build.sbt]: `ydoc-polyfill` it is

File: lib/java/ydoc-server/src/test/java/org/enso/ydoc/server/YdocTest.java (hunk 2 above)
Comment (4e6) [reply lib/java/ydoc-server/src/test/java/org/enso/ydoc/server/YdocTest.java:21]: They need to emulate initialization message exchange with the language server

**Review by JaroslavTulach — APPROVED** [review]
The description of the PR still contains:

> Ydoc into ydoc library and ydoc-server server parts

Otherwise I think this is good.

**Review by GregoryTravis — APPROVED** [review]

**Pull request merged**